up42/storage.py

```diff
@@ -168,27 +129,23 @@ def get_assets(
         Returns:
             A list of Asset objects.
         """
-        sort = f"{sortby},{'desc' if descending else 'asc'}"
-        url = f"{self.auth._endpoint()}/v2/assets?sort={sort}"
-        if created_before is not None:
-            url += f"&createdBefore={format_time(created_before)}"
-        if created_after is not None:
-            url += f"&createdAfter={format_time(created_after)}"
-        if workspace_id is not None:
-            url += f"&workspaceId={workspace_id}"
-        if collection_names is not None:
-            url += f"&collectionNames={collection_names}"
-        if producer_names is not None:
-            url += f"&producerNames={producer_names}"
-        if tags is not None:
-            for tag in tags:
-                url += f"&tags={tag}"
-        if sources is not None:
-            url += f"&sources={','.join(sources)}"
-        if search is not None:
-            url += f"&search={search}"
-
-        assets_json = self._query_paginated_endpoints(url=url, limit=limit)
+        params: AssetSearchParams = {
+            "createdAfter": format_time(created_after) if created_after is not None else None,
+            "createdBefore": format_time(created_before) if created_before is not None else None,
+            "workspaceId": workspace_id,
+            "collectionNames": collection_names,
+            "producerNames": producer_names,
+            "tags": tags,
+            "sources": sources,
+            "search": search,
+        }
+        assets_json = asset_search(
+            self.auth,
+            params_asset_search=params,
+            limit=limit,
+            sortby=sortby,
+            descending=descending,
+        )
 
         if (
```

Comment:
We either need to warn about deprecation or drop the parameters, but not warning and raising error at the same time, let's create a Jira ticket to clean this up

Reply:
https://up42.atlassian.net/browse/AT-4370?atlOrigin=eyJpIjoiZTRlODY1MTQxMzBiNDM1YmIwOWRhNzkzMDNlYzQ3ODAiLCJwIjoiaiJ9